### PR TITLE
[FIX] point_of_sale: undeletable order when it has fiscal position

### PR DESCRIPTION
**Pull request by caburj**
NOTE that this is also a refactor.

When an order has a fiscal position, deleting it from the ui is impossible
unless it's deleted from the localStorage. To reproduce:
- Make an order with fiscal position
- Validate the order, and in receipt screen, click next order.
  - This will remove the paid order from `pos`.
- [BUG] Refresh the page the product screen will focus on that order.
  - This is because the order is kept in the localStorage.

Why? `save_to_db` is dependent on `pos.selectedOrder` when there is a fiscal
position in the order.

How come? This is because when computing tax mapping, it's using the
`selectedOrder`. And `save_to_db` is triggered because when deleting an order
`selectedOrder` is set to a order.

The proposed change removes this dependency of an order to the
`pos.selectedOrder` and it fixes the issue.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by caburj** [comment]
@pimodoo Here is the refactoring of `get_taxes_after_fp`. It's main goal is to fix the undeletable order with fiscal position.

File: addons/point_of_sale/static/src/js/models.js [hunk 1]
```diff
@@ -1810,7 +1801,7 @@ class Orderline extends PosModel {
         return round_pr(this.get_unit_price() * this.get_quantity() * (1 - this.get_discount()/100), rounding);
     }
     get_taxes_after_fp(taxes_ids){
-        return this.pos.get_taxes_after_fp(taxes_ids);
+        return this.pos.get_taxes_after_fp(taxes_ids, this.order.fiscal_position);
```
Comment (pimodoo) [inline addons/point_of_sale/static/src/js/models.js:1804]: as we are in a refactoring in master, you can delete this function on posorderline and use the one on POS where this one was called/

Comment (caburj) [reply addons/point_of_sale/static/src/js/models.js]: Done. I added a temporary commit to highlight the change for this comment.

**Comment by pimodoo** [comment]
@caburj it looks good to me, you can squash the commits

**Comment by pimodoo** [comment]
@robodoo r+ rebase-ff

**Comment by caburj** [comment]
This needs retry, also, there is a linked PR in enterprise.

**Comment by pimodoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**